### PR TITLE
[Sketcher] Fix occasional crash after deleting last pole in b-spline creation mode

### PR DESCRIPTION
**Pull request by AjinkyaDahale**
Forum discussions: https://forum.freecadweb.org/viewtopic.php?f=10&t=66908, https://forum.freecadweb.org/viewtopic.php?f=10&t=66957.

### CODE REVIEW

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6528">pipeline status</a> for feature branch [PR_6528](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6528). Pipeline [486762224](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/486762224) was triggered at [6f5ba66](https://github.com/FreeCAD/FreeCAD/pull/6528/commits/6f5ba66fe330491433caf97101e256c70722dfd1). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6528">pipeline status</a> for feature branch [PR_6528](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6528). Pipeline [487444091](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/487444091) was triggered at [046e04d](https://github.com/FreeCAD/FreeCAD/pull/6528/commits/046e04dc113de2e56df26c4a960b6d6377cd556c). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6528">pipeline status</a> for feature branch [PR_6528](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6528). Pipeline [487733649](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/487733649) was triggered at [332bf39](https://github.com/FreeCAD/FreeCAD/pull/6528/commits/332bf39f46b92bd68e0b32d0dffd58f52c86a02c). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

File: src/Mod/Sketcher/Gui/DrawSketchHandler.cpp [hunk 1]
```diff
@@ -414,7 +414,10 @@ int DrawSketchHandler::seekAutoConstraint(std::vector<AutoConstraint> &suggested
         GeoId = preSelCrv;
         const Part::Geometry *geom = sketchgui->getSketchObject()->getGeometry(GeoId);
 
-        if(geom->getTypeId() == Part::GeomLineSegment::getClassTypeId()){
+        // ensure geom exists in case object was called before preselection is updated
+        if (!geom)
+            GeoId = GeoEnum::GeoUndef;
+        else if (geom->getTypeId() == Part::GeomLineSegment::getClassTypeId()) {
```
Comment (0penBrain) [inline src/Mod/Sketcher/Gui/DrawSketchHandler.cpp:420]: Most probably something like this is cleaner :
```
const Part::Geometry *geom = sketchgui->getSketchObject()->getGeometry(preSelCrv);
if (geom) {
    GeoId = preSelCrv;
    if (geom->getTypeId() == Part::GeomLineSegment::getClassTypeId()) {
        ...
    }
}
```

**Comment by abdullahtahiriyo** [comment]
I am going to merge it because it fixes an important bug. Then I am going to test it more in depth. I am tempted to do some small code changes separately, to better convey the intention of the code.

**Pull request merged**